android/src/main/kotlin/com/weplenish/flutter_wifi_connect/FlutterWifiConnectPlugin.kt

```diff
@@ -274,25 +277,37 @@ class FlutterWifiConnectPlugin() : FlutterPlugin, MethodCallHandler {
   @SuppressLint("MissingPermission")
   fun connect(@NonNull wifiConfiguration: WifiConfiguration, @NonNull result: Result){
     val network = wifiManager.addNetwork(wifiConfiguration)
+    if (network == -1) {
+      result.success(false)
+      return
+    }
     wifiManager.saveConfiguration()
-    wifiManager.disconnect()
 
     val wifiChangeReceiver = object : BroadcastReceiver() {
+      var count = 0
       override fun onReceive(context: Context, intent: Intent) {
+        count++;
         val info = intent.getParcelableExtra<NetworkInfo>(WifiManager.EXTRA_NETWORK_INFO)
-        if(info != null && info.isConnected){
-          result.success(info.extraInfo == wifiConfiguration.SSID || getSSID() == wifiConfiguration.SSID)
-          context?.unregisterReceiver(this)
+        if(info != null && info.isConnected) {
+          if (info.extraInfo == wifiConfiguration.SSID || getSSID() == wifiConfiguration.SSID) {
+            result.success(true)
+            context?.unregisterReceiver(this)
+          } else if (count > 1) {
+            // Ignore first callback if not success. It may be for the already connected SSID
+            result.success(false)
+            context?.unregisterReceiver(this)
+          }
         }
       }
     }
 
     val intentFilter = IntentFilter()
     intentFilter.addAction(WifiManager.NETWORK_STATE_CHANGED_ACTION)
     context?.registerReceiver(wifiChangeReceiver, intentFilter)
-    
+
+    // enable the new network and attempt to connect to it 
     wifiManager.enableNetwork(network, true)
-    wifiManager.reconnect()
```

Comment:
no need to reconnect, since we are setting to `attemptConnect` to `true` in `enableNetwork`